v2/examples/spiffe-tls/README.md

```diff
@@ -1,73 +1,81 @@
+One workload acts as a client and the other as the server. 
 
-The **server workload** creates a listener using the [spiffetls.Listen](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#Listen) function.
-It uses [spiffetls.MTLSServerWithSourceOptions](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#MTLSServerWithSourceOptions) to configure  the source, and validates that clients present an X509-SVID with an expected SPIFFE ID.
+The scenario goes like this:
+1. The server starts listening for incoming SPIFFE-compliant mTLS connections.
+2. The client establish an SPIFFE-compliant mTLS connection to the server. 
+3. The server starts waiting a message from the client.
+4. The client sends a "Hello server" message and starts waiting a response.
+5. The server reads the client's message, logs it to stdout, and sends a "Hello client" message as the response.
+6. The client reads the server's response and then closes the connection.
 
+## Listening
+To start listening for incoming connections the **server workload** uses the [spiffetls.ListenWithMode](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#ListenWithMode) function as follows:
 ```go
-listener, err := spiffetls.ListenWithMode(ctx, "tcp", serverAddress,
-    spiffetls.MTLSServerWithSourceOptions(
-        tlsconfig.AuthorizeID(clientID),
-        workloadapi.WithClientOptions(workloadapi.WithAddr(socketPath)),
-    ))
+	listener, err := spiffetls.ListenWithMode(ctx, "tcp", serverAddress,
+		spiffetls.MTLSServerWithSourceOptions(
+			tlsconfig.AuthorizeID(clientID),
+			workloadapi.WithClientOptions(workloadapi.WithAddr(socketPath)),
+		))
 ```
-
-Alternatively, the listener can use the `SPIFFE_ENDPOINT_SOCKET` environment variable to locate the Workload API.
-
+Where:
+- ctx is a `context.Context`. `ListenWithMode` blocks until the first Workload API response is received or this context times out or is cancelled.
+- serverAddress is the address (`localhost:55555`) where the server workload is going to listen for client connections.
+- [spiffetls.MTLSServerWithSourceOptions](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#MTLSServerWithSourceOptions) is used to configure the [X509Source](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2@v2.0.0-alpha.3/workloadapi?tab=doc#X509Source) used by the internal Workload API client.
+- clientID is a SPIFFE ID (`spiffe://example.org/client`), which along with the [tlsconfig.AuthorizeID](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#AuthorizeID) function makes the server to accept only clients that present an X509-SVID having such identity. You can pick any of the functions that return a [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) included with the library, or you can make your own. 
```

Comment:
function makes the server to accept only clients that present an X509-SVID having such identity -> function configures the server to accept only clients that present an X509-SVID with a matching SPIFFE ID

Here I'm not sure if I understood this correctly, so my suggestion may be wrong.

Reply:
You understood it correctly, your suggestion is just perfect, thanks!